src/test/mir-opt/enum_opt.rs

```diff
@@ -0,0 +1,42 @@
+// EMIT_MIR_FOR_EACH_BIT_WIDTH
+#![feature(arbitrary_enum_discriminant)]
+
+pub enum Uninhabited {
+  None,
+  Large([u64; 1024]),
+}
```

Comment:
Uninhabited types don't have any values, so either enum name is not right choice, or it is not testing what it is supposed to.

Reply:
mmm I just named it incorrectly. I changed it to `Niche`, which might also be wrong.

Reply:
Is this intended to test that enums using niche encoding aren't transformed? Could you add a short description of each test case to clarify what is being tested.  This enum isn't using a niche encoding currently.